### ISSUE TITLE
Feature Request: Sampled Softmax

### ISSUE THREAD

**Issue by Santosh-Gupta** [keras-team/keras#12558]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hello, 

This is a feature request to have a sampled softmax loss in Keras. The sampled softmax loss is an important loss function for many prominent ML architectures, particularly word embedding architectures. 

Technically, and function in tensorflow can should be able to done easily in Keras, but as far as I know, nobody has been successful in using Tensorflow's sampled softmax in keras. 

Here are the latest stackoverflow posts attempting to tackle this

https://stackoverflow.com/questions/49563614/tensorflows-sampling-softmax-slower-than-regular-softmax

https://stackoverflow.com/questions/53311701/tensorflow-multi-label-with-nce-or-sampled-softmax

https://stackoverflow.com/questions/52732145/sampled-softmax-in-tensorflow-keras

https://stackoverflow.com/questions/47892380/how-can-i-use-tensorflows-sampled-softmax-loss-function-in-a-keras-model

**Comment by abhipn** [comment]
+1 Any progress on this?

**Comment by ddofer** [comment]
Doesn't look like it. There's NCE and sampling in TF, but no idea how to use it neatly in Keras.

**Comment by kapilkd13** [comment]
Any update on this. I can try to raise a PR on it if someone can guide me about the changes that needs to be done
